sql-plugin/src/main/scala/com/nvidia/spark/rapids/spill/SpillFramework.scala

```diff
@@ -367,9 +367,14 @@ class SpillableHostBufferHandle private (
       }
     }
     if (materialized == null) {
-      materialized = closeOnExcept(HostMemoryBuffer.allocate(sizeInBytes)) { hmb =>
-        diskHandle.materializeToHostMemoryBuffer(hmb)
-        hmb
+      com.nvidia.spark.rapids.jni.RmmSpark.spillRangeStarting()
```

Comment:
Recording spills looks useful. From the API design perspective, I wonder if it would be better if `spillRangeStarting()` returned a `Closeable` that calls `spillRangeDone()` when it is closed. Perhaps this way is less error-prone as we could use some tool to flag the resources that are not properly closed.